app/src/main/java/com/example/dogbook/main/fragments/MapFragment.java

```diff
@@ -180,15 +185,31 @@ private void displayLocation() {
     }
 
     private void updateCoveredAreaBox() {
-        double longitudeLength = visibleAreaBox.northeast.longitude - visibleAreaBox.southwest.longitude;
-        double latitudeLength = visibleAreaBox.northeast.latitude - visibleAreaBox.southwest.latitude;
-        LatLng northeast = new LatLng(visibleAreaBox.northeast.latitude + (latitudeLength * 0.25), visibleAreaBox.northeast.longitude + (longitudeLength * 0.25));
-        LatLng southwest = new LatLng(visibleAreaBox.southwest.latitude - (latitudeLength * 0.25), visibleAreaBox.southwest.longitude - (longitudeLength * 0.25));
-        coveredAreaBox = new LatLngBounds(southwest, northeast);
+        double longitudeLength = abs(visibleAreaBox.northeast.longitude - visibleAreaBox.southwest.longitude);
+        double latitudeLength = abs(visibleAreaBox.northeast.latitude - visibleAreaBox.southwest.latitude);
+        float extraPercentageBySide = (COVERED_AREA_EXTRA_PERCENTAGE - 1)/2;
+
+        //Reduce percentage if the area exceeds the map
+        if (longitudeLength * COVERED_AREA_EXTRA_PERCENTAGE >= 360) {
+            extraPercentageBySide = (float) 0.25;
+        }
```

Comment:
Doc comments here would be useful to know why you need to do this -- why are you subtracting 1 for example from a percentage?